pkg/apis/certmanager/validation/issuer.go

```diff
@@ -179,6 +182,23 @@ func ValidateACMEIssuerDNS01Config(iss *v1alpha1.ACMEIssuerDNS01Config, fldPath
 				}
 			}
 		}
+		if p.Execute != nil {
+			if numProviders > 0 {
+				el = append(el, field.Forbidden(fldPath.Child("execute"), "may not specify more than one provider type"))
+			} else {
+				numProviders++
+				sks := &p.Execute.EnvSecret
+				secretFld := fldPath.Child("execute", "envSecretSecretRef")
+				if sks.Name == "" {
+					el = append(el, field.Required(secretFld.Child("name"), "secret name is required"))
+				}
+				if len(p.Execute.PluginName) == 0 {
+					el = append(el, field.Required(fldPath.Child("execute", "pluginName"), ""))
+				} else if match := executePluginNameRegex.MatchString(p.Execute.PluginName); !match {
+					el = append(el, field.Forbidden(fldPath.Child("execute", "pluginName"), "pluginName field contains invalid characters"))
```

Comment:
Might be better off shadowing `fldPath` with `fldPath.Child("execute")` to avoid accidentally missing it in one of these lines 😄